middleware/RetryHandler.md

```diff
@@ -10,7 +10,11 @@ Provide a reusuable component that provides application developers with effectiv
 - A `retry-attempt` header SHOULD be added which contains a numeric value representing the retry number.  This value will be used for diagnostics and determining the effectiveness of the retry handler.
 - Retries MUST respect the `retry-after` response header if provided.
 - Where no `retry-after` header is provided by the server, an exponential backoff with random offset hueristic should be used to determine the retry delay.
-- Retries should be limited to a maximum value.
+- Retries should be limited to a maximum delay value. The default value for this is set at 180 seconds.
+- The client code can specify a custom value for the maximum delay.
+- Retries will be based on the cumulative retry delay compared against the maximum delay specified by the client code.
+- Upon expiry of the maximum delay, when the cumulative retry delay is greater than the specified maximum delay, the `Task` should be cancelled and an `exception` thrown with a relevant message.
+- In the case whereby the client code specifies a maximum delay that is less than the received `retry-after` header value from the server, the cumulative retry delay will run for the duration of the `retry-after` value.
```

Comment:
If the client code specifies a maximum delay less than the retry-after, then we should respect  the client's choice and fail despite the service indicating a retry-after.

Reply:
@MIchaelMainer by failing, do you propose we fail fast and do not even attempt to delay the retry request as specified by the `retry-after` or _exponential back-off_ values? Or do we enter the `Delay()` `Task` and retry at least once then fail? (see my [comment](https://github.com/microsoftgraph/msgraph-sdk-design/pull/18/commits/93bc4d2bc56a81aba2e44677a8cffb3d23dff5e8#r293300084) on Peter's review).

Reply:
Good question. I need to be more specific. 
* If the retry-after exceeds the max delay, we should fail fast. The service is telling us to wait longer than value max delay submitted by the dev's code. No need to wait until we reach the max delay time.
* If the exponential back-off values exceeds the max delay value, I think we could wait to retry once more at the max delay time, and then fail if that fails.

No retry should occur after reaching the max delay time. Customer specified should always be conservatively respected.